### PR TITLE
Remove ICU4J

### PR DESCRIPTION
**Pull request by dmikurube**
#1570 started to use standard plugins with `embulk-util-guess:0.3.0`, which has its own copy from ICU4J. Embulk is now free from ICU4J that has occupied ~10MB in the executable binary.

The embedded `lib/embulk/guess/charset.rb` needed ICU4J, but it's now obsolete. The standard guess plugins should have no problems because they have their own copy of ICU4J. Some third-party guess plugins may be impacted, though.

### CODE REVIEW

**Review by hiroyuki-sato — APPROVED** [review]
Basically, LGTM 👍 I found a typo. (missing space)

File: embulk-core/src/main/java/org/embulk/deps/guess/CharsetMatch.java [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
+@Deprecated
+public class CharsetMatch {
+    private CharsetMatch() {
+        // No direct instantiation.
+    }
 
-    public abstract String getName();
+    @Deprecated
+    public int getConfidence() {
+        return 50;
```
[the remark this answers is not in the document]
Comment (dmikurube) [reply embulk-core/src/main/java/org/embulk/deps/guess/CharsetMatch.java:11]: Yes. It comes from the context of ICU4J's `CharsetMatch`.
https://unicode-org.github.io/icu-docs/apidoc/dev/icu4j/com/ibm/icu/text/CharsetMatch.html#getConfidence--

The number is just a placeholder to fake, then not very important, but I wanted to show the `UTF-8` is just a dummy guess.

File: embulk-core/src/main/java/org/embulk/deps/guess/CharsetDetector.java [hunk 2]
```diff
@@ -1,51 +1,31 @@
...
-        try {
-            return (Class<CharsetDetector>) CLASS_LOADER.loadClass(CLASS_NAME);
-        } catch (final ClassNotFoundException ex) {
-            throw new LinkageError("Dependencies for Guess are not loaded correctly: " + CLASS_NAME, ex);
-        }
+    @Deprecated
+    public static CharsetDetector create() {
+        logger.warn(
+                "Class org.embulk.deps.guess.CharsetDetector is no longer available. "
+                + "It always returns \"UTF-8\" unconditionally.Use an appropriate guess plugin explicitly.");
```
[the remark this answers is not in the document]
Comment (dmikurube) [reply embulk-core/src/main/java/org/embulk/deps/guess/CharsetDetector.java]: Thanks! Fixed in 05f5ee8dbcbf77fabe23b3ba72455700e93b7d15

**Comment by dmikurube** [comment]
Thanks!

**Pull request merged**